File: src/main/java/my/study/repository/design/pattern/proxy/ProxyMen.java
package my.study.repository.design.pattern.proxy;

/**
 * @author djh on  2018/9/7 13:44
 * @E-Mail 1544579459@qq.com
 */
public class ProxyMen implements Subject {

    private RealImp mRealImp;

    ProxyMen(RealImp realImp) {
        mRealImp = realImp;
    }

    @Override
    public void someMethod() {
        mRealImp.someMethod();
    }
}

File: src/main/java/my/study/repository/design/pattern/proxy/kotlin/ProxyMen.kt
package my.study.repository.design.pattern.proxy.kotlin

/**
 * @author djh on  2018/10/15 19:26
 * @E-Mail 1544579459@qq.com
 */
class ProxyMen
(private val subject: Subject = RealImp()) : Subject {
    
    override fun someMethod() {
        subject.someMethod()
    }
}